### PR TITLE
Thermometer settings added

### LINKED ISSUE #1756: Adding the settings menu for thermometer
**Actual Behaviour**

No settings for thermometer yet.

**Expected Behaviour**

Settings menu should be there for thermometer where -
1. User can choose unit
2. User can choose which sensor to use.

**Steps to reproduce it**
N/A

**LogCat for the issue**

N/A

**Screenshots of the issue**

N/A

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by ho-dor**
Fixes #1756 

**Changes**: Settings menu with the option to change units.

**Screenshot/s for the changes**:
thermo_gif

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[app-debug.zip](https://github.com/fossasia/pslab-android/files/3243490/app-debug.zip)

### CODE REVIEW

**Comment by ho-dor** [comment]
@CloudyPadmal @neel1998 please review.

File: app/src/main/java/io/pslab/fragment/ThermometerDataFragment.java [hunk 1]
```diff
@@ -578,4 +583,26 @@ private void initiateThermoSensor(int type) {
...
+    public void setUnit(){
+        if("F".equals(ThermometerDataFragment.unit)){
+            currentMax = 257;
+            currentMin = -40;
+            thermometer.setMaxSpeed(PreferenceManager.getDefaultSharedPreferences(getActivity()).getFloat(thermoSensor.THERMOMETER_MAX_LIMIT, 257));
+            thermometer.setMinSpeed(PreferenceManager.getDefaultSharedPreferences(getActivity()).getFloat(thermoSensor.THERMOMETER_MIN_LIMIT,-40));
+            label_statAvg.setText(R.string.avg_thermo_fahrenheit);
+            label_statMax.setText(R.string.max_thermo_fahrenheit);
+            label_statMin.setText(R.string.min_thermo_fahrenheit);
+            thermometer.setUnit("F");
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/fragment/ThermometerDataFragment.java]: Put the degree sign to this as well similar to C

File: app/src/main/java/io/pslab/fragment/ThermometerSettingsFragment.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+import io.pslab.R;
+import io.pslab.others.PSLabPermission;
+
+public class ThermometerSettingsFragment extends PreferenceFragmentCompat implements SharedPreferences.OnSharedPreferenceChangeListener {
+
+    public static final String KEY_INCLUDE_LOCATION = "include_location_sensor_data";
+    public static final String KEY_UPDATE_PERIOD = "setting_thermo_update_period";
+    public static final String KEY_THERMO_SENSOR_TYPE = "setting_thermo_sensor_type";
+    public static final String KEY_THERMO_SENSOR_GAIN = "setting_thermo_sensor_gain";
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/fragment/ThermometerSettingsFragment.java]: We don't have a "Gain" for thermo sensor. Can remove this setting.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -310,12 +310,27 @@
         <item>TSL2561</item>
     </string-array>
 
+    <string-array name="thermo_sensor_list">
+        <item>In-built Sensor</item>
+        <item>SHT21</item>
+    </string-array>
+
+    <string-array name="thermo_unit_list">
+        <item>F</item>
```
Comment (CloudyPadmal) [inline app/src/main/res/values/strings.xml]: Degree sign

**Comment by ho-dor** [comment]
@CloudyPadmal check

**Comment by neel1998** [comment]
@ho-dor  when the user opens the configuration for the first time. No unit is shown in select unit preference. Please cet C as default. Also add space in the menu title "Thermometer Confguration"

**Comment by ho-dor** [comment]
> @ho-dor when the user opens the configuration for the first time. No unit is shown in select unit preference. Please cet C as default. Also add space in the menu title "Thermometer Confguration"

The menu title is actually consistent with all the other instruments.

**Comment by ho-dor** [comment]
I have the updated the default unit value . Please check the updated apk.

**Pull request merged**